examples/multicall_busybox.rs

```diff
@@ -5,27 +5,39 @@ use clap::{App, AppSettings, Arg};
 fn main() {
     let app = App::new(env!("CARGO_CRATE_NAME"))
         .setting(AppSettings::ArgRequiredElseHelp)
-        .arg(
-            Arg::new("install")
-                .long("install")
-                .help("Install hardlinks for all subcommands in path")
-                .exclusive(true)
-                .takes_value(true)
-                .default_missing_value("/usr/local/bin")
-                .use_delimiter(false),
+        .subcommand(
+            App::new("busybox")
+                .arg(
+                    Arg::new("install")
+                        .long("install")
+                        .help("Install hardlinks for all subcommands in path")
+                        .exclusive(true)
+                        .takes_value(true)
+                        .default_missing_value("/usr/local/bin")
+                        .use_delimiter(false),
+                )
+                .subcommand(App::new("true").about("does nothing successfully"))
+                .subcommand(App::new("false").about("does nothing unsuccessfully")),
         )
         .subcommand(App::new("true").about("does nothing successfully"))
         .subcommand(App::new("false").about("does nothing unsuccessfully"));
```

Comment:
Should we splut out the double-add into a function so we show reuse, which people are more likely to do?

Reply:
I'm not sure I understand what you mean. Something like this to deduplicate the added commands, or a separate function called twice to ensure they don't get out of sync.

i.e. this:

```rust
let mut busybox = App::new("busybox").arg(
    Arg::new("install")
        .long("install")
        .help("Install hardlinks for all subcommands in path")
        .exclusive(true)
        .takes_value(true)
        .default_missing_value("/usr/local/bin")
        .use_delimiter(false),
);
let mut app = App::new(env!("CARGO_CRATE_NAME"))
    .setting(AppSettings::ArgRequiredElseHelp)
    .setting(AppSettings::Multicall)
    .subcommand_value_name("APPLET")
    .subcommand_help_heading("APPLETS");

for a in [&mut busybox, &mut app] {
    a.subcommand(App::new("true").about("does nothing successfully"))
        .subcommand(App::new("false").about("does nothing unsuccessfully"));
}
app.subcommand(applet);
```

or this:

```rust
fn true_applet() -> App<'static> {
    App::new("true").about("does nothing successfully");
}

fn false_applet() -> App<'static> {
    App::new("false").about("does nothing unsuccessfully");
}

let mut app = App::new(env!("CARGO_CRATE_NAME"))
    .setting(AppSettings::ArgRequiredElseHelp)
    .setting(AppSettings::Multicall)
    .subcommand_value_name("APPLET")
    .subcommand_help_heading("APPLETS")
    .subcommand(
        App::new("busybox")
            .arg(
                Arg::new("install")
                    .long("install")
                    .help("Install hardlinks for all subcommands in path")
                    .exclusive(true)
                    .takes_value(true)
                    .default_missing_value("/usr/local/bin")
                    .use_delimiter(false),
            )
            .subcommand(true_applet())
            .subcommand(false_applet()),
    )
    .subcommand(true_applet())
    .subcommand(false_applet());
```